src/components/organisms/AssetActions/Edit/EditAdvancedSettings.tsx

```diff
@@ -103,6 +117,15 @@ export default function EditAdvancedSettings({
         Logger.error(content.form.error)
         return
       } else {
+        if (price.type === 'free') {
+          const tx = await setMinterToDispenser(
+            ocean,
+            ddo.dataToken,
+            accountId,
+            setError
+          )
+          if (!tx) return
```

Comment:
so what does user see in case of failed transaction? If I see the code right, you simply keep on showing the loading screen?

Reply:
It will show updating ddo failed
![image](https://user-images.githubusercontent.com/39853992/123225657-818d3580-d505-11eb-8692-cc9b4b6bf921.png)

Reply:
This error message looks ok to me. @kremalicious were there any other outstanding issues with this PR?